### PR TITLE
test(site): move users page test to storybook

### LINKED ISSUE #14535: flake: `UsersPage.test.tsx`
https://github.com/coder/coder/actions/runs/10680634527/job/29602816185

```
FAIL src/pages/UsersPage/UsersPage.test.tsx (19.402 s)
  ● UsersPage › suspend user › when it is success › shows a success message

...
      24 | 	const user = userEvent.setup();
      25 | 	// Get the first user in the table
    > 26 | 	const moreButtons = await screen.findAllByLabelText("More options");
         | 	                                 ^
      27 | 	const firstMoreButton = moreButtons[0];
      28 | 	await user.click(firstMoreButton);
      29 |

      at waitForWrapper (node_modules/.pnpm/@testing-library+dom@9.3.3/node_modules/@testing-library/dom/dist/wait-for.js:162:27)
      at node_modules/.pnpm/@testing-library+dom@9.3.3/node_modules/@testing-library/dom/dist/query-helpers.js:86:31
      at findAllByLabelText (src/pages/UsersPage/UsersPage.test.tsx:26:35)
      at Object.suspendUser (src/pages/UsersPage/UsersPage.test.tsx:141:11)
```

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
By using [Storybook interaction tests](https://storybook.js.org/tutorials/ui-testing-handbook/react/en/interaction-testing/), we improve the developer experience, as we can see what is happening in the browser, use browser tools to debug, and snapshot the component after interactions, making our tests more reliable and easier to maintain.

Fixes https://github.com/coder/coder/issues/14535

### CODE REVIEW

File: site/src/pages/UsersPage/storybookData/users.ts [hunk 1]
```diff
@@ -0,0 +1,577 @@
+import type { LoginType, User, UserStatus } from "api/typesGenerated";
```
Comment (code-asher) [inline site/src/pages/UsersPage/storybookData/users.ts:1]: Should we replace all the emails with fake ones?  Some of these are probably real (the Coder ones at least, maybe the ones from ChatGPT too).

File: site/src/pages/UsersPage/storybookData/groups.ts [hunk 2]
```diff
@@ -0,0 +1,546 @@
...
+
+function findMockedUserById(id: string): User | undefined {
+	const user = MockUsers.find((user) => user.id === id);
+	return user;
+}
+
+// These values were retrieved from the Coder API. Sensitive information such as
+// usernames, names, and emails has been replaced with fake user data to protect
+// privacy. This user data comes from the MockUsers to keep consistency.
+export const MockGroups: Group[] = [
```
Comment (code-asher) [inline site/src/pages/UsersPage/storybookData/groups.ts:12]: Should we replace/remove the ones with external avatar URLs?  Could be weird if our tests are making requests for these external jpgs and such

File: site/src/pages/UsersPage/UsersPage.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,397 @@
...
+		if (!userRow) {
+			throw new Error("No user row found");
+		}
+		spyOn(API, "updateUserRoles").mockRejectedValue({});
+
+		await user.click(within(userRow).getByLabelText("Edit user roles"));
+		await user.click(
+			within(userRow).getByLabelText("Auditor", { exact: false }),
+		);
+		await within(document.body).findByText("Error updating the user roles.");
```
Comment (code-asher) [inline site/src/pages/UsersPage/UsersPage.stories.tsx:391]: Looks like the error text is `Error on updating the user roles.`

**Review by code-asher — APPROVED** [review]
Love to see tests replaced with Storybook!

File: site/src/pages/UsersPage/storybookData/users.ts (hunk 1 above)
Comment (BrunoQuaresma) [reply site/src/pages/UsersPage/storybookData/users.ts:1]: Ah, sorry, I thought I had removed all of them. 🤦

**Pull request merged**